wcfsetup/install/files/lib/system/service/worker/Util.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace wcf\system\service\worker;
+
+use Jose\Component\Core\JWK;
+use ParagonIE\ConstantTime\Base64UrlSafe;
+use ParagonIE\ConstantTime\Binary;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class Util
+{
+    /**
+     * Unserialize the public key into its x and y components.
+     *
+     * @param string $data
+     * @return array{x: string, y: string}
+     */
+    public static function unserializePublicKey(string $data): array
+    {
+        $data = \bin2hex($data);
+        if (Binary::safeSubstr($data, 0, 2) !== '04') {
+            throw new \InvalidArgumentException('Invalid public key format');
+        }
+        $data = Binary::safeSubstr($data, 2);
+        $dataLength = Binary::safeStrlen($data);
+
+        return [
+            'x' => \hex2bin(Binary::safeSubstr($data, 0, $dataLength / 2)),
+            'y' => \hex2bin(Binary::safeSubstr($data, $dataLength / 2)),
+        ];
```

Comment:
I don't understand the purpose of the hexadecimal encoding here. I believe the logic would be much simpler and easier to follow if it directly dealt with the binary data.

Side note: `Binary::safe*()` is not required here. The misfeature this protects against was removed in PHP 8.0.

Untested:

```suggestion
        if ($data[0] !== "\x04") {
            throw new \InvalidArgumentException('Invalid public key format');
        }
        $data = \substr($data, 1);
        $dataLength = \strlen($data);
        if ($dataLength !== /* whatever length is expected */) {
            throw new \InvalidArgumentException('Invalid public key format');
        }
        
        [$x, $y] = \str_split($data, \intdiv($dataLength, 2));

        return [
            'x' => $x,
            'y' => $y,
        ];
```